### Assignment 3:Reproduce the analysis of the Hard to Employ program in NY:

In [17]:
import os
import sys
import numpy as np
import pylab as pl
import json
s = json.load(open(os.getenv("PUIDATA")+"/fbb_matplotlibrc.json") )
pl.rcParams.update(s)
%pylab inline

Populating the interactive namespace from numpy and matplotlib


### Null hypothesis: the % of former prisoners recidivised after release is the same or higher for candidates who participated in the program as for the control group,significence level p=0.05.  
    
#### $H_0: p_1 >= p_0 $                               
#### $H_1: p_1 < p_0 $



In [9]:
alpha=0.05
P_0=11.7*0.01
P_1=10.0*0.01
if P_0-P_1 <= 0:
    # we are done
    print ("the Null holds")
else:
    print ("we must assess the statistical significance")
n_0 = 409
n_1 = 568

#lets get the counts by multiplying by the sample size
Nt_0 = P_0 * n_0
Nt_1 = P_1 * n_1


we must assess the statistical significance


In [10]:
#define the sample proportion first
sp = (P_0 * n_0 + P_1 * n_1) / (n_1 + n_0)
print (sp)

0.107116683726


In [11]:
p = lambda p0, p1, n0, n1: (p0 * n0 + p1 * n1) / (n0 + n1)
#standard error
se = lambda p, n0, n1: np.sqrt(p * (1 - p) * (1.0 / n0 + 1.0 / n1)) 

In [12]:
zscore = lambda p0, p1, s : (p0 - p1) / s
z_3y = zscore(P_1, P_0, se(p(P_0, P_1, n_0, n_1), n_0, n_1))
print (z_3y)

-0.84764020522


In [13]:
p_3y=1-0.7995

#report functino from prof's code
def report_result(p,a):
    print ('is the p value ' + 
           '{0:.2f} smaller than the critical value {1:.2f}?'.format(p,a))
    if p < a:
        print ("YES!")
    else: 
        print ("NO!")
    
    print ('the Null hypothesis is {}'.format(\
                            'rejected' if p < a  else 'not rejected') )

    
report_result(p_3y, alpha)

is the p value 0.20 smaller than the critical value 0.05?
NO!
the Null hypothesis is not rejected


In [15]:
def evalChisq(values):
    values = np.array(values)
    E = np.empty_like(values)
    for j in range(len(values[0])):
        for i in range(2):
            
            E[i][j] = ((values[i,:].sum() * values[:,j].sum()) / 
                        (values).sum())
    return ((values - E)**2 / E).sum()

# $\chi^2 $ Test 

$\chi^2 = \sum_i \frac{(f_{i,observed} - f_{i,expectated})^2}{f_{i,expected}}$

In [20]:
Ntot = 977 # a + b + c + d = tot

sample_values = np.array([[0.10*568, 0.90*568], [0.117*409,(1-0.117)*409]])
print (evalChisq(sample_values))

0.718493917505


DOF is 1 here. 
From the table, the threshold is 3.84. The chi square statistics is 0.718, which is smaller than 3.84.
Therefore we could not reject null hypothesis.

<pre>
```

| convicted of a fellony |  Yes   |  No   |      |
|------------------------|--------|-------|------|
|   test sample          |  56.8  | 511.2 |  568 |
|------------------------|--------|-------|------|
|   control sample       | 47.853 |361.147|  409 |
|------------------------|--------|-------|------|
|         total          | 104.653|872.347|  977 |
```
</pre>

For "Ever employed in a CEO transitional job", the rejection of Null hypothesis for z test is the absolute value of z statistics not less than 1.65 and for Chi square test is the chi square statistics larger than 3.84. 


For the "Convicted of a felony after 3 years data",
the rejection region of Null hypothesis for z test is th absolute value of z score not less than 1.65; for Chi square test is the chi square statistics larger than 3.84. 